model_analyzer/cli/cli.py

```diff
@@ -12,241 +12,64 @@
+                    parser_args['action'] == 'store_true'
+                    or parser_args['action'] == 'store_false'):
+                self._parser.add_argument(
+                    *config.flags(),
+                    default=argparse.SUPPRESS,
+                    help=config.description(),
+                    **config.parser_args(),
+                )
+            else:
+                self._parser.add_argument(
+                    *config.flags(),
+                    default=argparse.SUPPRESS,
+                    choices=config.choices(),
+                    help=config.description(),
+                    type=config.field_type(),
+                    **config.parser_args(),
                 )
-                args.export_path = None
-            elif args.export_path and not os.path.isdir(args.export_path):
-                raise TritonModelAnalyzerException(
-                    f"Export path {args.export_path} is not a directory.")
-        if args.triton_launch_mode == 'remote':
-            if args.client_protocol == 'http' and not args.triton_http_endpoint:
-                raise TritonModelAnalyzerException(
-                    "client-protocol is 'http'. Must specify triton-http-endpoint "
-                    "if connecting to already running server or change protocol using "
-                    "--client-protocol.")
-            if args.client_protocol == 'grpc' and not args.triton_grpc_endpoint:
-                raise TritonModelAnalyzerException(
-                    "client-protocol is 'grpc'. Must specify triton-grpc-endpoint "
-                    "if connecting to already running server or change protocol using "
-                    "--client-protocol.")
-        args.gpus = args.gpus.split(',')
-
-    def _setup_logger(self, args):
-        """
-        Setup logger format
-        """
-        log_level = logging.getLevelName(args.log_level)
-        logging.basicConfig(level=log_level,
-                            format="%(asctime)s.%(msecs)d %(levelname)-4s"
-                            "[%(filename)s:%(lineno)d] %(message)s",
-                            datefmt="%Y-%m-%d %H:%M:%S")
 
     def parse(self):
         """
         Retrieves the arguments from the command line and loads them into an
-        ArgumentParser Also does some sanity checks for arguments.
-
-        Returns
-        -------
-        argparse.Namespace
-            containing all the parsed arguments
-
-        Raises
-        ------
-        TritonModelAnalyzerException
-            For arguments passed incorrectly
+        ArgumentParser. It will also configure the Model Analyzer config
+        accordingly.
         """
 
         # Remove the first argument which is the program name
         args = self._parser.parse_args(sys.argv[1:])
-        self._preprocess_and_verify_arguments(args)
-        self._setup_logger(args)
-
-        return args
+        self._config.set_config_values(args)
```

Comment:
How does this work. If I pass a config file AND command line arguments, do the CLI args override the settings in the config file? What if elements are missing from the config file, is there a default value?

Reply:
@dzier Yes, exactly. The CLI args have a higher priority, then config values, and then the default value. There is a test for this too. 
https://github.com/triton-inference-server/model_analyzer/pull/58/files#diff-82fdafeb870e92fb6c61f571ad81add4eb2173aa56f521175cab4139f3015baeR320-R330